updated status doc to represent all completed and planned K01 functional requirements

couryrr-afs opened the pull request:
## Describe your changes

It appears that some of the status were out of data, missing or incorrect. I wanted to update the document to be as accurate as possible for K01.

## Issue ticket number and link

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [ ] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

gberardi-pillar committed:
Use dependency injection for ChargePoint.

Signed-off-by: Gianfranco Berardi <54074967+gberardi-pillar@users.noreply.github.com>

Commit:
updated status doc to represent all completed and planned K01 function requirements
updated some comment and test names after community discussion.

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>

christopher-davis-afs commented on doc/ocpp_201_status.md:
```diff
@@ -1225,58 +1224,57 @@ This document contains the status of which OCPP 2.0.1 numbered functional requir
-| K01.FR.11 |        |                                                                                                 |
-| K01.FR.12 |        |                                                                                                 |
-| K01.FR.13 |        |                                                                                                 |
+| K01.FR.10 | ⛽️K08  |                                                                                                 |
```
There's still stuff relevant for libocpp in this one, and we enforce it in how we store the profiles

couryrr-afs replied:
I added a comment about the values being set during validation. I still think that it is accurate that the bulk of the work is in K08 for this though.

marcemmers committed:
Fix usage of SSL default password callback (#696)

* Fix usage of SSL default password callback

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Added length verifications

Signed-off-by: AssemblyJohn <ioan.bogdann@gmail.com>

---------

Signed-off-by: Marc Emmers <m.emmers@alfen.com>
Signed-off-by: AssemblyJohn <ioan.bogdann@gmail.com>
Co-authored-by: AssemblyJohn <ioan.bogdann@gmail.com>

AssemblyJohn committed:
Update API for new library header usage (#717)

* Updated for new header usage
* Updated deps
---------

Signed-off-by: AssemblyJohn <ioan.bogdann@gmail.com>

AssemblyJohn committed:
Replaced throws with error handling (#718)

Signed-off-by: AssemblyJohn <ioan.bogdann@gmail.com>

Pietfried committed:
added support for SummaryInventory in OCPP201 (#716)

Signed-off-by: pietfried <pietgoempel@gmail.com>

Pietfried committed:
adjust some component schmeas and configs to align with requirements of OCPP201 spec (#720)

Signed-off-by: pietfried <pietgoempel@gmail.com>

Pietfried committed:
Refactored OCPP16 error handling in state machine (#673)

* Refactored OCPP16 error handling in state machine:
* changed ChargePoint interface: only two functions on_error and on_error_cleared remain
* these two functions can be used to report and clear informational errors (no change to Faulted state) or fault errors
* ErrorInfo struct was introduced to have a compact datatype for the ErrorInformation used in a StatusNotification.req
* Added tests for error handling in ChargePointFSM
* StatusNotification.req triggered by TriggerMessage.req could contain wrong information before the added changes
* Added more simple fault handling to OCPP201 as well
---------

Signed-off-by: pietfried <pietgoempel@gmail.com>
Signed-off-by: Piet Gömpel <37657534+Pietfried@users.noreply.github.com>

Pietfried committed:
Allow resuming of transactions for OCPP1.6 (#704)

* Allow resuming of transactions for OCPP1.6:
* charge_point.start() function can now receives a set of active transactions
* If an internal transaction in libocpp can be identified from the internal database and the session_id is also listed in the set of active transactions, libocpp will resume the transaction
* If an internal transaction in libocpp can be identified from the internal database and the session_id is not listed in the set of active transactions, libocpp will initiate a StopTransaction.req
---------

Signed-off-by: pietfried <pietgoempel@gmail.com>

Pietfried committed:
revert hardcoded csms uri in config (#724)

Signed-off-by: pietfried <pietgoempel@gmail.com>

kromahn committed:
Fix missing include (#726)

* Fix missing include

When building under alpine, it fails due to following issue: error: "'int32_t' does not name a type". Including header <cstdint> fixes it.

Signed-off-by: Katja Romahn <37145145+kromahn@users.noreply.github.com>

* Fix include order

Signed-off-by: Katja Romahn <37145145+kromahn@users.noreply.github.com>

---------

Signed-off-by: Katja Romahn <37145145+kromahn@users.noreply.github.com>

hikinggrass committed:
Bump libocpp version to 0.15.0, bump libevse-security to 0.8.0 (#732)

* Bump libevse-security dependency version to 0.8
* Bum libocpp version to 0.15.0

---------

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

Pietfried committed:
moving add_stop_energy_wh (#737)

Signed-off-by: pietfried <pietgoempel@gmail.com>

christopher-davis-afs committed:
Implement SetChargingProfileRequest and basic database storage/loading (#711)

* smart_charging: Make profile and evse_id order consistent

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* charge_point: Make `handle_message` protected

This makes it possible to use from our unit tests.

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* charge_point: Add SmartChargingHandler

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* smart_charging: Make SmartChargingHandler mockable

Also add the mock class for it.

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* charge_point: Handle SetChargingProfileRequest

Add a handler for SetChargingProfileRequest with unit tests that
ensure that we add valid profiles - and only valid profiles.

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* doc: Update SmartCharging status

Add functional requirements handled by the implementation of
`handle_set_charging_profile_req()` and a few missed FRs covered
by `validate_profile()`.

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* charge_point: Test that set_charging_profile callback is called

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* added ability to load profile from database

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>
Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* charge_point: Reject ChargingStationExternalConstraints profiles in SetChargingProfileRequest

Implements K01.FR.22

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* updated database statements for tests

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>

* added columns to database and updated equality checks

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>

* smart_charging: Add combined validation and add method

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* Updated FR45 to use supplyPhases

Signed-off-by: Peter Giavotto <146003699+Giavotto@users.noreply.github.com>

* database_handler: Correct spelling in database tests

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* database_handler: Test profile equality

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* database_handler: Rename function to get all profiles grouped by evse id

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>

* moved charing profile equality to test since it is not a full comparison

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>

* removed duplicate include

Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>

---------

Signed-off-by: Christopher Davis <150722105+christopher-davis-afs@users.noreply.github.com>
Signed-off-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>
Signed-off-by: Peter Giavotto <146003699+Giavotto@users.noreply.github.com>
Co-authored-by: Coury Richards <146002925+couryrr-afs@users.noreply.github.com>
Co-authored-by: Peter Giavotto <146003699+Giavotto@users.noreply.github.com>

marcemmers committed:
Update generators and move custom enums to new file (#741)

* Update code_generator files, rename to ocpp_enums.hpp so we have room for custom enums in enums.hpp

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Run generators and replace all instances of enums.hpp to ocpp_enums.hpp

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Reintroduce custom enum types, fix typo

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Remove use of RevolutionsPerMinute since thats not an official unit

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Fix linting

Signed-off-by: Marc Emmers <m.emmers@alfen.com>

* Fix issue in ocpp_types.cpp.jinja: A type of std::optional<std::vector<Enum>> was not converted to and from its enum value

Signed-off-by: pietfried <pietgoempel@gmail.com>

---------

Signed-off-by: Marc Emmers <m.emmers@alfen.com>
Signed-off-by: pietfried <pietgoempel@gmail.com>
Co-authored-by: pietfried <pietgoempel@gmail.com>
